### PR TITLE
Remove experiment property from logger base class.

### LINKED ISSUE #11234: [RFC] Deprecate `LightningLoggerBase.experiment` requirement from Logger interface
## Proposed refactor

Deprecate the `experiment` property required by the Logger interface

https://github.com/PyTorchLightning/pytorch-lightning/blob/dedfde6859a15270385e6e751567b29ab6488e87/pytorch_lightning/loggers/base.py#L104-L107

### Motivation

1. `experiment` carries no meaning as specified on the base logger. This is an object which can be anything. 

Currently, Lightning appears to promise users that `logger.experiment` will have a consistent interface across different logger implementations, which is not true!

If users are interacting with the `experiment` directly, then they must *already know* what Logger implementation they're using. 

A good rule of thumb is that the only required interfaces for the Logger are what the Trainer directly calls. Everything else can be pushed to various implementations.  Since the trainer does not interact with the `experiment` at all, it does not need to be required on the Logger API. 

What makes more sense is for the various logger implementations to directly offer public property on their interface. For example, the TensorBoard logger could expose the `SummaryWriter` as a property in its public interface for Tensorboard logger users to call in their own code. 

2. Authors of custom loggers need to define this seemingly random experiment property without knowing how/when it will be called.

### Pitch

1. Make the current `experiment` definitions in the logger implementations top-level properties in their respective classes.
2. Deprecate `experiment` off the base logger API

### Additional context

_Originally posted by @ananthsub in https://github.com/PyTorchLightning/pytorch-lightning/issues/11209#issuecomment-999393358_

Part of #7740

cc @justusschock @awaelchli @akihironitta @tchaton @borda @edward-io @ananthsub

### PR DESCRIPTION
**Pull request by akashkw**
## What does this PR do?
Remove experiment from abstract class. We will still leave experiment in our logger implementations, but we will no longer make that a requirement.

Fixes #11234 

## Before submitting

- [X] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [X] Did you write any **new necessary tests**? (not for typos and docs)
- [X] Did you verify new and **existing tests pass** locally with your changes?
- [X] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -281,6 +281,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated function `pytorch_lightning.callbacks.device_stats_monitor.prefix_metric_keys` ([#11254](https://github.com/PyTorchLightning/pytorch-lightning/pull/11254))
 
 
+- Deprecated the abstract property `pytorch_lightning.loggers.LightningLoggerBase.experiment` ([#11234](https://github.com/PyTorchLightning/pytorch-lightning/pull/11603))
```
Comment (rohitgr7) [inline CHANGELOG.md:284]:
```suggestion
- Removed the abstract property `LightningLoggerBase.experiment` ([#11603](https://github.com/PyTorchLightning/pytorch-lightning/pull/11603))
```
removed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -281,6 +281,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated function `pytorch_lightning.callbacks.device_stats_monitor.prefix_metric_keys` ([#11254](https://github.com/PyTorchLightning/pytorch-lightning/pull/11254))
 
 
+- Removed the abstract property `LightningLoggerBase.experiment` ([#11603](https://github.com/PyTorchLightning/pytorch-lightning/pull/11603))
```
Comment (daniellepintz) [inline CHANGELOG.md:284]: please also move to "Removed" section just below

Comment (daniellepintz) [reply CHANGELOG.md:284]: also please update title of the PR accordingly

File: CHANGELOG.md [hunk 3]
```diff
@@ -419,7 +419,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `Strategy.on_tpu` property ([#11536](https://github.com/PyTorchLightning/pytorch-lightning/pull/11536))
 
 
- - Removed access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
+- Removed access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
```
Comment (daniellepintz) [inline CHANGELOG.md:422]: why is this added? I think you may need to merge with master

Comment (akashkw) [reply CHANGELOG.md:422]: This was just fixing an indentation error. There was a random space at the start of the line above my changelog message.

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by daniellepintz — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/loggers/base.py [hunk 4]
```diff
@@ -101,11 +101,6 @@ def update_agg_funcs(
         if agg_default_func:
             self._agg_default_func = agg_default_func
 
-    @property
-    @abstractmethod
-    def experiment(self) -> Any:
```
Comment (tchaton) [inline pytorch_lightning/loggers/base.py:106]: @carmocca Should we have a depreciation there?

Comment (carmocca) [reply pytorch_lightning/loggers/base.py:106]: This was an `abstractmethod` so all subclasses were forced to override it.

Which means that removing it from the parent should be backwards compatible as each subclass will still have it available.

**Comment by daniellepintz** [comment]
@akashkw if you merge with master the test failures should go away

**Comment by daniellepintz** [comment]
Can someone enable auto-merge on this please?

**Comment by daniellepintz** [comment]
Looks like we need a rerun on the TPU tests :/ they seem to be quite flaky. Can someone with access help out?
